spark/src/main/scala/io/delta/sharing/spark/DeltaSharingOptions.scala

```diff
@@ -0,0 +1,162 @@
+ * limitations under the License.
+ */
+
+package io.delta.sharing.spark
+
+// scalastyle:off import.ordering.noEmptyLine
+import java.util.Locale
+
+import scala.util.Try
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.network.util.{ByteUnit, JavaUtils}
+import org.apache.spark.sql.catalyst.util.CaseInsensitiveMap
+
+trait DeltaSharingOptionParser {
+  protected def options: CaseInsensitiveMap[String]
+
+  def toBoolean(input: String, name: String): Boolean = {
+    Try(input.toBoolean).toOption.getOrElse {
+      throw DeltaSharingErrors.illegalDeltaOptionException(name, input, "must be 'true' or 'false'")
+    }
+  }
+}
+
+trait DeltaSharingReadOptions extends DeltaSharingOptionParser {
+  import DeltaSharingOptions._
+
+  val maxFilesPerTrigger = options.get(MAX_FILES_PER_TRIGGER_OPTION).map { str =>
+    Try(str.toInt).toOption.filter(_ > 0).getOrElse {
+      throw DeltaSharingErrors.illegalDeltaOptionException(
+        MAX_FILES_PER_TRIGGER_OPTION, str, "must be a positive integer")
+    }
+  }
+
+  val maxBytesPerTrigger = options.get(MAX_BYTES_PER_TRIGGER_OPTION).map { str =>
+    Try(JavaUtils.byteStringAs(str, ByteUnit.BYTE)).toOption.filter(_ > 0).getOrElse {
+      throw DeltaSharingErrors.illegalDeltaOptionException(
+        MAX_BYTES_PER_TRIGGER_OPTION, str, "must be a size configuration such as '10g'")
+    }
+  }
+
+  val ignoreChanges = options.get(IGNORE_CHANGES_OPTION).exists(toBoolean(_, IGNORE_CHANGES_OPTION))
+
+  val ignoreDeletes = options.get(IGNORE_DELETES_OPTION).exists(toBoolean(_, IGNORE_DELETES_OPTION))
+
+  val readChangeFeed = options.get(CDF_READ_OPTION).exists(toBoolean(_, CDF_READ_OPTION)) ||
+    options.get(CDF_READ_OPTION_LEGACY).exists(toBoolean(_, CDF_READ_OPTION_LEGACY))
+
+  val startingVersion: Option[DeltaStartingVersion] = options.get(STARTING_VERSION_OPTION).map {
+    case "latest" => StartingVersionLatest
+    case str =>
+      Try(str.toLong).toOption.filter(_ >= 0).map(StartingVersion).getOrElse{
+        throw DeltaSharingErrors.illegalDeltaOptionException(
+          STARTING_VERSION_OPTION, str, "must be greater than or equal to zero")
+      }
+  }
+
+  val startingTimestamp = options.get(STARTING_TIMESTAMP_OPTION)
+
+  val cdfOptions: Map[String, String] = prepareCdfOptions()
+
+  val versionAsOf = options.get(TIME_TRAVEL_VERSION).map { str =>
+    Try(str.toLong).toOption.filter(_ > 0).getOrElse {
+      throw DeltaSharingErrors.illegalDeltaOptionException(
+        TIME_TRAVEL_VERSION, str, "must be a positive integer")
+    }
+  }
+
+  val timestampAsOf = options.get(TIME_TRAVEL_TIMESTAMP)
+
+  private def prepareCdfOptions(): Map[String, String] = {
+    if (readChangeFeed) {
+      validCdfOptions.filter(option => options.contains(option._1)).map(option =>
+        option._1 -> options.get(option._1).get
+      )
+    } else {
+     Map.empty[String, String]
+    }
+  }
+
+  private def provideOneStartingOption(): Unit = {
```

Comment:
Nit: consider calling this: validateOneStartingOption

Reply:
updated.